src/array.jl

```diff
@@ -453,6 +453,23 @@ function Base.unsafe_convert(::Type{CuDeviceArray{T,N,AS.Global}}, a::DenseCuArr
                                a.maxsize - a.offset*Base.elsize(a))
 end
 
+## synchronization behavior
+
+"""
+    unsafe_disable_task_sync!(arr::CuArray)
+
+By default `CuArray`s are implicitly synchronized when they are used on different CUDA streams.
+A `CuArray` that is used on multiple Julia tasks will be used by different streams
+and thus will cause a synchronization between multiple Julia tasks.
+
+This `unsafe_disable_task_sync` disables synchronization when stream are being switched.
+"""
+function unsafe_disable_task_sync!(arr::CuArray)
+    return arr.data[].task_sync = false
```

Comment:
Could one think about having `unsafe_disable_task_sync!` to return the `arr` object instead of (?) the flag? This would allow e.g. using the piping operator
```
julia> a = CUDA.rand(2, 1) |> CUDA.unsafe_disable_task_sync!
false
```
and thus be useful when using the feature upon array creation.

Reply:
And similar for `unsafe_enable_task_sync!`